homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -93,14 +97,178 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
+                        // add listen addresses to kademlia routing table
+                        for addr in info.listen_addrs {
+                            behavior.kademlia.add_address(&peer_id, addr);
+                        }
+                    }
+
+                    // rendezvous
+                    // we are good to register self & discover with any node we contact. more peers = more better!
+                    if info.protocols.contains(&RENDEZVOUS_PROTOCOL_NAME) {
+                        // register self with remote
+                        if let Err(err) = behavior.rendezvous_client.register(
+                            Namespace::from_static(RENDEZVOUS_NAMESPACE),
+                            peer_id,
+                            None,
+                        ) {
+                            warn!(
+                                err = format!("{err}"),
+                                peer_id = peer_id.to_string(),
+                                "failed to register with rendezvous peer"
+                            )
+                        }
+                        // discover other nodes
+                        behavior.rendezvous_client.discover(
+                            Some(Namespace::from_static(RENDEZVOUS_NAMESPACE)),
+                            None,
+                            None,
+                            peer_id,
+                        );
+                    }
+                }
+                identify::Event::Pushed { peer_id } => debug!(
+                    peer_id = peer_id.to_string(),
+                    "pushed identify info too peer"
+                ),
+            }
+        }
+        SwarmEvent::Behaviour(ComposedEvent::RendezvousClient(rendezvous_client_event)) => {
+            match rendezvous_client_event {
+                rendezvous::client::Event::Discovered {
+                    rendezvous_node,
+                    registrations,
+                    cookie,
+                } => {
+                    // save cookie for later (when we are hungry for snacks again. yummy.)
+                    if cookie.namespace() == Some(&Namespace::from_static(RENDEZVOUS_NAMESPACE)) {
+                        event_handler
+                            .rendezvous_cookies
+                            .insert(rendezvous_node, cookie);
+
+                        // dial discovered peers
+                        for registration in registrations {
+                            // TODO: do anything with ttl here?
+                            let opts = DialOpts::peer_id(registration.record.peer_id())
+                                .addresses(registration.record.addresses().to_vec())
+                                .condition(libp2p::swarm::dial_opts::PeerCondition::Disconnected)
+                                .build();
+                            // TODO: we might be dialing too many peers here. Add settings to configure when we stop dialing new peers
```

Comment:
Important to note, since rendezvous can give us hundreds of nodes and without limits we will dial _all_ of them, which we then will identify and dial all of those, leading to exponentially more dials until we try to dial all the nodes that use homestar. We are only working with small networks atm so this isnt horrible yet, and this PR is getting fairly beefy already so imo should be in a new PR.

Reply:
@mriise makes sense. Can you open an enhancement issue to that we create some thresholding around this?